modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/LoginIdentity.kt

```diff
@@ -0,0 +1,16 @@
+package au.com.shiftyjelly.pocketcasts.repositories.sync
+
+sealed class LoginIdentity(val value: String) {
+    object PocketCasts : LoginIdentity("PocketCasts")
+    object Google : LoginIdentity("Google")
+
+    companion object {
+        fun valueOf(value: String?): LoginIdentity {
+            return when (value) {
+                PocketCasts.value -> PocketCasts
+                Google.value -> Google
+                else -> PocketCasts
+            }
+        }
```

Comment:
I just tried to switch this over but it seems like enums already have a valueOf function.

```
>>> enum class LoginIdentity(val value: String) {
...     POCKET_CASTS("PocketCasts"),
...     GOOGLE("Google")
... }
>>>
>>> LoginIdentity.valueOf("Google")
java.lang.IllegalArgumentException: No enum constant Line_4.LoginIdentity.Google
	at java.base/java.lang.Enum.valueOf(Enum.java:274)
	at Line_4$LoginIdentity.valueOf(Line_4.kts)
	at Line_6.<init>(Line_6.kts:1)

>>> LoginIdentity.valueOf("GOOGLE")
res4: Line_4.LoginIdentity = GOOGLE
```
With other enums we have a `fromString` method to handle this. For example: 
```
        enum class ResultsType(val value: String) {
            PODCASTS("podcasts"),
            EPISODES("episodes"),
            UNKNOWN("unknown");
            companion object {
                fun fromString(value: String?) =
                    ResultsType.values().find { it.value == value } ?: UNKNOWN
            }
        }
```        
I do like the idea of not having to add extra options later. 🤔 
It is possible to do the following with a sealed class to get the sub classes but it requires reflection.
```
LoginIdentity::class.sealedSubclasses
```
I'm not sure what to do so I have left it for now.

> Also, if we hit the else case, could it be helpful to log an error message or maybe even crash on debug builds?

My thinking was if the user upgraded to this version and was already logged in, this should automatically get the login identity Pocket Casts. I have changed the code to return a null and push the decision to the calling code if the value isn't found, rather than hiding it in the `valueOf` method.
This made me realise I didn't add test steps to ensure that upgrading to this version doesn't sign the user out. I have tested this case and my account remains logged in during the upgrade.

Reply:
> My thinking was if the user upgraded to this version and was already logged in, this should automatically get the login identity Pocket Casts. I have changed the code to return a null and push the decision to the calling code if the value isn't found, rather than hiding it in the valueOf method.

Makes sense. I didn't realize that this could happen during a normal flow. I thought it would only come up if we added a new type and forgot to handle it in the valueOf method.

> This made me realise I didn't add test steps to ensure that upgrading to this version doesn't sign the user out. I have tested this case and my account remains logged in during the upgrade.

Good catch. I tested this as well and didn't observe any issues. 👍